### ISSUE TITLE
BUG: scipy.interpolate interp1d extrapolate behaviour change from 1.8.1 -> 1.9.0

### ISSUE THREAD

**Issue by TM90** [scipy/scipy#16813]
### Describe your issue.

In scipy 1.9.0 the interp1d method with 'previous' kind and fill_value='extrapolate' seems to use the largest positive value for extrapolation. 

While in 1.8.1 the last value was used for extrapolation (This seems to be the right behaviour)

To reproduce the issue execute the example code with scipy 1.9.0 and 1.8.1

Plot of the below code example in 1.8.1
![interp1dextrapolate_1 8 1](https://user-images.githubusercontent.com/5484864/183686446-2112977f-8cef-49ae-acea-d87a9814db19.png)

And in 1.9.0
![interp1dextrapolate_1 9](https://user-images.githubusercontent.com/5484864/183686468-a30f6592-be10-428a-8499-05b0f96d5a82.png)

### Reproducing Code Example

```python
import numpy as np
from scipy.interpolate import interp1d
from matplotlib.pyplot import figure, show


x = np.arange(5)
y = np.array([0.0, 0.0, 1.0, 0.0, 0.0])

new_x = np.arange(50) / 10.
new_y = interp1d(x, y, 'previous', bounds_error=False, fill_value="extrapolate")(new_x)

f = figure()
s = f.add_subplot(111)
s.plot(x, y, 'x')
s.plot(new_x, new_y)
show()
```

### Error message

```shell
/
```

### SciPy/NumPy/Python version information

1.9.0 1.23.1 sys.version_info(major=3, minor=9, micro=5, releaselevel='final', serial=0)

**Comment by ev-br** [comment]
Cc @AtsushiSakai , what would you say? Also cf https://github.com/scipy/scipy/issues/9591

**Comment by AtsushiSakai** [comment]
Ah... I think my PR #16128 looks very bad.
I don't not why I used `max` or ` min` for extrapolation.
https://github.com/scipy/scipy/blob/649932fa64747ef585eabe75cf9a39e3799e3eab/scipy/interpolate/_interpolate.py#L533
https://github.com/scipy/scipy/blob/649932fa64747ef585eabe75cf9a39e3799e3eab/scipy/interpolate/_interpolate.py#L542
I think it should be the last or first element of `y`
I will fix it soon and the RP should be included in v1.9.1.
I'm sorry for everyone.

**Comment by ev-br** [comment]
A case of collective blindness, more like. When you send a fix, could you test both with and without `assume_sorted`.
